packages/react-aria-components/docs/advanced.mdx

```diff
@@ -0,0 +1,345 @@
+  return `Selected date: ${formatted}`;
+}
+```
+
+This enables a `<CalendarValue>` to be placed inside a `<Calendar>` to display the current value.
+
+```tsx
+<Calendar>
+  {/* ... */}
+  {/*- begin highlight -*/}
+  <CalendarValue />
+  {/*- end highlight -*/}
+</Calendar>
+```
+
+The state interfaces and their associated contexts supported by each component are listed in the Advanced Customization section of their documentation.
+
+## Hooks
+
+If you need to customize things even further, such as accessing internal state, intercepting events, or customizing DOM structure, you can drop down to the lower level Hook-based API. Hooks only provide behavior and leave all rendering to you. This gives you more control and flexibility, but requires additional glue code to set up.
+
+React Aria Components and Hooks can be used together, allowing you to mix and match depending on the level of customization you require. We recommend starting with the component API by default, and only dropping down to hooks when you need to customize something that the component API does not allow.
+
+Some potential usecases for Hooks are:
+
+* Overriding which DOM element a component renders
+* Rendering a subset of the children (e.g. virtualized scrolling)
+* Intercepting a DOM event to apply conditional logic
+* Overriding internal state management behavior
+* Customizing overlay positioning
+* Removing unused features to reduce bundle size
+
+### Setup
+
+As described [above](#contexts), each React Aria component exports a corresponding context. You can build a custom implementation of a component using Hooks by consuming from the relevant context with <TypeLink links={docs.links} type={docs.exports.useContextProps} />.
+
+This example shows how a custom checkbox could be set up using `CheckboxContext` from `react-aria-components` and the [useCheckbox](useCheckbox.html) hook from `react-aria`.
+
+```tsx
+import type {CheckboxProps} from 'react-aria-components';
+import {CheckboxContext, useContextProps} from 'react-aria-components';
+import {useToggleState} from 'react-stately';
+import {useCheckbox} from 'react-aria';
+
+const MyCheckbox = React.forwardRef((props: CheckboxProps, ref: React.ForwardedRef<HTMLInputElement>) => {
+  // Merge the local props and ref with the ones provided via context.
+  ///- begin highlight -///
+  [props, ref] = useContextProps(props, ref, CheckboxContext);
+  ///- end highlight -///
+
+  // Follow the hook docs...
+  let state = useToggleState(props);
+  let {inputProps} = useCheckbox(props, state, ref);
+  return <input {...inputProps} ref={ref} />;
+});
+```
+
+Since `MyCheckbox` consumes from `CheckboxContext`, it can be used within other React Aria Components in place of the default `Checkbox` component, such as within a [Table](Table.html) or [GridList](GridList.html). This lets you replace only the checkbox with a custom implementation, without rewriting all other React Aria Components you might use it in.
```

Comment:
```suggestion
Since `MyCheckbox` consumes from `CheckboxContext`, it can be used within other React Aria Components in place of the default `Checkbox` component, such as the ones found within a [Table](Table.html) or [GridList](GridList.html). This lets you specifically customize the checkbox without rewriting all other React Aria Components you might use.
```